source/guides/basics/loops.md

```diff
@@ -46,7 +139,74 @@ my_move_task:
         - wait 2s
 ```
 
-You can try this in-game via `/ex run my_move_task`. It will repeatedly ask you to move until you move at least 3 blocks away.
+You can try this in-game via `/ex run my_move_task`. This task when run will keep telling you to move away until you have moved 3 blocks away from when you first were.
+
+Just like the `repeat` command we have a `wait` command at the end of the loop. While it can be optional for `repeat` loops to use this, it is almost always required for `while` loops. Without it it will try to check as much as it can all at the same time, which can cause the server to crash.
+
+As a word of warning, `while` loops should be avoided if possible. It is very easy to make what is known as an infinite loop, as in the loop will never have a chance to stop. Once its running, it is very hard to force it to stop beyond just stopping the server.
+
+### Stop The Loop
+
+Sometimes in loops you only want to keep looping until you reach a certain point, however not necessarily stop the script. While `stop` is used to stop scripts entirely, `repeat/foreach/while stop` is used to stop the relevant loop
+
+For instance in the first example we had, if we wanted to stop sending lightning if the player had less than 5 health we would do as follows
+
+```dscript_green
+my_lightning_task:
+    type: task
+    script:
+    - repeat 5 as:count:
+        - strike <player.location>
+        - narrate "you have been struck by lightning <[count]> times!"
+        - if <player.health> < 5:
+            - repeat stop
+        - wait 1s
+    - narrate "no more lightning"
+```
+Take note, only the loop stops, the end `narrate` line still runs.
+
+It can also be used in `foreach` loops. For instance, if we wanted to stop looping if we found a baby cow in the example above
+
+```dscript_green
+my_cow_task:
+    type: task
+    script:
+    - foreach <player.location.find_entities[cow].within[30]> as:cow:
+        - narrate "<[loop_index]> - There's a cow just <[cow].location.distance[<player.location>].round> blocks away!"
+        - playeffect effect:fireworks_spark at:<[cow].location> visibility:50 quantity:100 data:0 offset:3
+        - if <[cow].is_baby>:
+            - narrate "omg its a baby!!"
+            - foreach stop
+```
```

Comment:
Unnecessary as it is said that it works for foreach and while loops

Reply:
I thought it would be good to demonstrate in multiple examples, at the very least to make it clear of the syntax